doc/visualizer-in-depth.md

```diff
@@ -0,0 +1,524 @@
+# A p5 visualizer, in detail
+
+As on the [previous page](visualizer-basics.md), let's begin with a (more
+detailed) diagram of the life cycle of a visualizer. This version represents
+all of the methods that your Visualizer class might want to implement, extend,
+or override to accomplish a variety of effects, as well as all of the
+significant actions that might occur as someone explores sequences with your
+visualizer.
+
+````d2 layout="elk"
```

Comment:
I think the new diagrams look great!  The colours are a huge help!  Except I thought all the parameter change etc. was going to come from dots instead of from draw?

Reply:
I had a paragraph about that in my TL;DR post about the diagram above. If we did that, we'd lose the contrast that the three events (change sequence, resize canvas, change parameter) all _disrupt_ the draw loop, which can't restart til you get back around to it in the diagram by following that path of arrows, as opposed to a mouse event which can just happen and the draw loop keeps chugging along as it was. So that's why I left those three coming out of draw, and then to address Aaron's concern that it then looked like the draw() method would call or cause one of those things to happen, I changed it to an "other code section" node and explained that it calls draw() for you. Is that OK? It seemed like the best way to capture the contrast between these three events and all of the other key press, touch, accelerometer, etc. events.

Reply:
Thanks, I missed that!  Makes sense.